controlplane/eks/api/v1beta1/awsmanagedcontrolplane_types.go

```diff
@@ -166,6 +166,14 @@ type AWSManagedControlPlaneSpec struct { //nolint: maligned
 	// Amazon VPC CNI addon or if you have specified a secondary CIDR block.
 	// +kubebuilder:default=false
 	DisableVPCCNI bool `json:"disableVPCCNI,omitempty"`
+
+	// DisableKubeProxy indicates that kube-proxy should be disabled. With EKS clusters kube-proxy
+	// is automatically installed into the cluster. For clusters where you want to use kube-proxy
+	// functionality that is provided with an alternate CNI, this option provides a way to specify
+	// that the kube-proxy daemonset should be deleted. You cannot set this to true if you are using
+	// the Amazon kube-proxy addon.
+	// +kubebuilder:default=false
+	DisableKubeProxy bool `json:"disableKubeProxy,omitempty"`
```

Comment:
As this will live only in v1beta1 we will need to update the conversion functions

Reply:
Ok, I'd be appreciative if you could point me to an example in the conversion code where the same situation exists (exists in v1beta1 but not in earlier revisions)

Reply:
You can refer to [the PR](https://github.com/kubernetes-sigs/cluster-api-provider-aws/pull/3337) where updateConfig field is added to AWSManagedMachinePool type, specifically */conversion.go.

Reply:
Conversion routines updated in v1alpha3 & v1alpha4 and make test is successful.